util/printer/workflow-printer.go

```diff
@@ -137,7 +137,7 @@ func countPendingRunningCompletedNodes(wf *wfv1.Workflow) (int, int, int) {
 	running := 0
 	completed := 0
 	for _, node := range wf.Status.Nodes {
-		tmpl := wf.GetTemplateByName(node.TemplateName)
+		tmpl := wf.GetTemplateByName(util.GetTemplateFromNode(node))
 		if tmpl == nil || !tmpl.IsPodType() {
 			continue
 		}
```

Comment:
I don't know if the math would work out entirely the same, but this could potentially be simplified to just check `node.Type` instead of getting the template only to get its type

Reply:
Hrm which types should we be checking for?

```
NodeTypePod       NodeType = "Pod"
NodeTypeContainer NodeType = "Container"
NodeTypeSteps     NodeType = "Steps"
NodeTypeStepGroup NodeType = "StepGroup"
NodeTypeDAG       NodeType = "DAG"
NodeTypeTaskGroup NodeType = "TaskGroup"
NodeTypeRetry     NodeType = "Retry"
NodeTypeSkipped   NodeType = "Skipped"
NodeTypeSuspend   NodeType = "Suspend"
NodeTypeHTTP      NodeType = "HTTP"
NodeTypePlugin    NodeType = "Plugin"
```

Reply:
Correct. If the number is the same, then it shouldn't matter. I'm not sure if it's entirely semantically equivalent though

Reply:
Neither am I and there is no test coverage for this code. I'll try running this change locally with debugging and verify. It might take some time

Reply:
Oh I didn't realize this didn't have a unit test. It does seem to have coverage via other CLI commands, just not its _own_ unit test.

In that case, that simplification can be done as a separate PR (if possible, or just tests if not would be great too!). The change you made here is still correct.

Reply:
> The existing check is `!tmpl.IsPodType()`, so I imagine `node.Type != NodeTypePod` would be equivalent

ok well actually this did not need much verification -- the function just above `IsPodType`, `GetNodeType`, has this [exact conversion logic](https://github.com/argoproj/argo-workflows/blob/3f9b992cdddb09633ef329c13eab1823ca6e5f3e/pkg/apis/workflow/v1alpha1/workflow_types.go#L3030). 

It would be `node.Type != wfv1.NodeTypePod` in this file specifically.

> Neither am I and there is no test coverage for the lines I'm changing.

Also there is an existing test case that covers this, the [`--wide` test](https://github.com/argoproj/argo-workflows/blob/3f9b992cdddb09633ef329c13eab1823ca6e5f3e/util/printer/workflow-printer_test.go#L87) ("P/R/C" = "Pending/Running/Completed"). More unit tests or test cases wouldn't hurt though

Reply:
Simplified in #13291